http/http.go

```diff
@@ -0,0 +1,88 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// Response is the type that all the HTTP requests are going to return.
+type Response struct {
+	Code    int
+	Body    interface{}
+	Headers http.Header
+}
+
+//Request is the
+type Request struct {
+	URL     string
+	Headers map[string]string
+	Body    []byte
+}
+
+// Get executes a GET request against the specified URL.
+func Get(r Request, out interface{}) (*Response, error) {
+	return do("GET", r, out)
+}
+
+// Post executes a POST request against the specified URL.
+func Post(r Request, out interface{}) (*Response, error) {
+	return do("POST", r, out)
+}
+
+// Patch executes a PATCH request against the specified URL.
+func Patch(r Request, out interface{}) (*Response, error) {
+	return do("PATCH", r, out)
+}
+
+// Put executes a PUT request against the specified URL.
+func Put(r Request, out interface{}) (*Response, error) {
+	return do("PUT", r, out)
+}
+
+func do(method string, request Request, out interface{}) (*Response, error) {
+
+	// Build the request
+	req, err := http.NewRequest(method, request.URL, bytes.NewBuffer(request.Body))
+	if err != nil {
+		return nil, fmt.Errorf("Error building the %s request for %s: %v", method, request.URL, err)
+	}
+
+	// Fill the request with the headers
+	for k, v := range request.Headers {
+		req.Header.Set(k, v)
+	}
+	client := &http.Client{}
+	client.Timeout = time.Second * time.Duration(10)
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Error doing %s to %s: %v", method, request.URL, err)
+	}
+	defer resp.Body.Close()
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading the response when doiing %s to %s: %v", method, request.URL, err)
```

Comment:
errors should start with lowercase :)

Reply:
why not use errors.New() too?

Reply:
Didn't know about it.

Reply:
I'm using it to format the output, checking the docs looks like it's the way to go https://golang.org/pkg/errors/#example_New_errorf

Is there any reason to use `errors.New` over `ftm.Errorf`?